### ISSUE TITLE
Controller is not working after Kubernetes all Cluster node is down 

### ISSUE THREAD

**Issue by dungnt081191** [uber/uReplicator#287]
@Technoboy-  @yangy0000  
Hi . here i have an dissaster case . If all node of my kubernetes cluster is down . 
All Worker pod can not graceful shutdown , Controller is not working and restart .
During this time , uReplicator still replicate from Source to Destination . 
What happen when Kubernetes is down , all node ? - uReplicator is deploying in Kubernetes GCP .

I tested and this is result : 
- Controller is not working anymore
- Worker is restarting 1 time , and don't  replicate any message anymore , Destination topic received correct message from Source to Destination before Cluster is down.  ALL message after uReplicator down is in lag ( in consumer-group at Source cluster ). Message is keep rising in Source message ( all become lag ) 
- NOTHING WORK until I restart Controller by manual delete Controller pod.
- After restart by manually delete Controller pod , Controller assign again to Worker ,and message replicate correctly .

the QUESTION is : 
1 - Can Controller now restart automatically?  , if yes, how long have i must to wait ?. 
2 - What happen if Controller is down for a long time ? 
3 - How much message Controller can assign to Worker ? - unlimited ?
4 - Do anyone have more detail config , run for non/federate mode? the current document is very very unuseful you know .
5 - best pratice in uReplicator in PRODUCTION ? - number of Controller / Worker , time duration have to wait after restart , rebalance replicate message if 1 worker down ....?

**Comment by Technoboy-** [comment]
1. controller can not restart automatically, you can do this via shell . so in production, you'd better deploy more than one controller, and uReplicator will response for switching controller when one is down.
2. if Controller is down , workers can still replicate message. but if some workers down during this period,  the left alive worker can still replicate message of the partitions allocated by controller， that means some partitions message will not replicate anymore.
3. it can calculate by partition. 1500 partition up or down. and it also depends the amount of message in the partition and worker load. you should monitor the metric worker logged, such as lags to decide add worker.
4. sorry for my poor English, but we will do more in documents
5. @yangy0000 can help .

**Comment by dungnt081191** [comment]
@Technoboy-  do you have skypeId , i wanna call for clear lots of things , it's save so so much of time :D

**Comment by dungnt081191** [comment]
@yangy0000 
do you have any idea about make Controller restart after Kubernetes was down ?

**Comment by Technoboy-** [comment]
could you startup more than one controller, as one is down, other will take the responsible for the cluster.
if you want to restart controller as k8s restart, you have to write shell as kill worker .

**Comment by dungnt081191** [comment]
@Technoboy-  yeah , i started 2 Controller . But let's me clear something about helix server.
Controller pod is running , Controller service is running but after a long time ( about 24h alive ) .
Helix server in Controller is not running anymore ( curl localhost:9000/topics **FAILED** ).

 How to restart it automatically?

**Comment by Technoboy-** [comment]
what do you mean is not running anymore ? does another controller take the responsibility of the failed one ?
if you wanna restart it automatically , you have to write some task like shell to ping the controller, after continuous like more than 5 times not responsible,  you have to kill the pod , and restart new one .
